datatypes/src/lib.rs

```diff
@@ -1,3 +1,22 @@
+// configure default clippy lints
+#![deny(clippy::correctness)]
+#![warn(clippy::complexity, clippy::style, clippy::perf, clippy::pedantic)]
+// disable some pedantic lints
+#![allow(
+    clippy::cast_possible_truncation,
+    clippy::cast_possible_wrap,
+    clippy::cast_precision_loss,
+    clippy::cast_sign_loss,
+    clippy::default_trait_access,
+    clippy::module_name_repetitions,
+    clippy::must_use_candidate,
+    clippy::non_ascii_literal,
+    clippy::similar_names,
+    clippy::single_match_else,
+    clippy::type_repetition_in_bounds,
+    clippy::wildcard_imports
```

Comment:
`warning: unknown clippy lint: clippy::wildcard_imports`

Reply:
I would also allow `missing_errors_doc` as I think this will often be straightforward and result in boiler doc

Reply:
`map_flatten` and `filter_map` could also be allowed

Reply:
it's fine for me to fail CI when there are linter errors